### ISSUE TITLE
Question: why is :443/console being used as a health check for master ELBs?

### ISSUE THREAD

**Issue by ryanrf** [aws-quickstart/quickstart-redhat-openshift#181]
Sorry if this is the wrong place for this. If it is, please let me know where I can ask general questions like this.

I'm curious why the health check for the ELB in front of the OpenShift Masters is using `:443/console/` as a health check instead of just `:443`, which points to the API server. This means if the webconsole crashes and that health check is failing the masters will be recreated, even though the actual master services (api, scheduler, controllers) are still up.

I ran into this recently when I modified the webconsole's configMap and set values that prevented the webconsole pods from starting correctly. This meant the ELB thought the entire master instances were down, which was definitely not the case. All other services were still up.

**Comment by pierreozoux** [comment]
@ryanrf then in your opinion, which end point should we test?

**Comment by pierreozoux** [comment]
We also have to keep in mind #107 during this conversation.

What is your opinion @jaymccon ?
If the QS aims to be an implementation of the reference architecture, we should foolow there defaults. But maybe there was a good reason to choose otherwise.

**Comment by jaymccon** [comment]
It would be great to get a pr to move the healthchecks to /api @ryanrf. I do seem to remember that there were some issues with this, when I tried it initially, but am a bit fuzzy on the details. Definitely worth a re-look.

There are a few more benefits to the healthchecks/asg method. If a single master instance fails (and it's not [0]) the cluster can automatically recover. It also allows easy scale out by adjusting the desired count in the asg. With those in mind I'd be hesitant to get rid of autoscaling for the Masters/Etcd, but some technique to prevent all the masters from being blown away at the same time is definitely needed. 

I also need to check whether the bug in #107 is still present, if so we should just manually copy the needed files across during bootstrapping, to ensure all masters have the needed files.

**Comment by ryanrf** [comment]
I submitted the PR here: https://github.com/aws-quickstart/quickstart-redhat-openshift/pull/183

I should add that I'm only using version 3.9, so I'm can't confirm any issues with 3.7. 

@jaymccon I completely agree that ASGs with the master are awesome. It makes putting this into an infrastructure pipeline fairly straight forward, encouraging code reviews/ unit tests for even infrastructure code.
If a user does choose to put this into a pipeline and they want to scale the master instances up the current `AllowedValues` setting for those parameters prevents the instances number from being changed. In my above PR I've also added MinValue to the master and etcd instances, allowing them to be scaled up.
